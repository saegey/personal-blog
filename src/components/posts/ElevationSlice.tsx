import { Box, Grid, Text } from 'theme-ui'
import { gradeToColor, formatTime } from '../../lib/formatters'
import { useUnits } from '../../context/UnitProvider'

const ElevationSlice = ({ marker }: any) => {
  const units = useUnits()

  return (
    <Grid
      gap={2}
      columns={[2, 4, 4]}
      sx={{
        padding: '10px',
        borderLeftColor: 'mutedAccent',
        borderLeftStyle: 'solid',
        borderLeftWidth: '1px',
        borderRightColor: 'mutedAccent',
        borderRightStyle: 'solid',
        borderRightWidth: '1px',
      }}
    >
      <Box>
        <Text as="p">Grade</Text>
        <Text
          sx={{
            fontSize: '20px',
            color: marker ? gradeToColor(marker.grade * 100) : 'black',
          }}
        >
          {marker && marker.grade ? `${(marker.grade * 100).toFixed(1)}%` : '-'}
        </Text>
      </Box>
      <Box>
        <Text as="p">Distance</Text>
        <Text sx={{ fontSize: '20px' }}>
          {marker && marker.distance
            ? `${marker.distance} ${units.distanceUnit}`
            : '-'}
        </Text>
      </Box>
      <Box>
        <Text as="p">Elevation</Text>
        <Text sx={{ fontSize: '20px' }}>
          {marker && marker.y
            ? `${marker.y.toFixed(0)} ${units.elevationUnit}`
            : '-'}
        </Text>
      </Box>
      <Box>
        <Text as="p">Time</Text>
        <Text sx={{ fontSize: '20px' }}>
          {marker && marker.x ? `${formatTime(marker.x)}` : '-'}
        </Text>
      </Box>
    </Grid>
  )
}

export default ElevationSlice
