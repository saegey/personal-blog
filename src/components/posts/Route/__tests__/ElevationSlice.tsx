/**
 * @jest-environment jsdom
 */
import renderer from 'react-test-renderer'

import ElevationSlice from '../ElevationSlice'

describe('ElevationSlice', () => {
  it('renders', () => {
    const json = renderer.create(<ElevationSlice />).toJSON()
    expect(json).toMatchSnapshot()
  })

  it('renders values', () => {
    const json = renderer
      .create(
        <ElevationSlice
          marker={{ grade: 0.04, distance: 100, y: 50.4, x: 1200 }}
        />
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })
})
