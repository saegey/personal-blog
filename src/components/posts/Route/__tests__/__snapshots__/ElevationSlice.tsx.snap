// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElevationSlice renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: grid;
  grid-gap: 8px;
  grid-template-columns: repeat(2, 1fr);
  padding: 10px;
  border-left-color: mutedAccent;
  border-left-style: solid;
  border-left-width: 1px;
  border-right-color: mutedAccent;
  border-right-style: solid;
  border-right-width: 1px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    grid-template-columns: repeat(4, 1fr);
  }
}

@media screen and (min-width: 52em) {
  .emotion-0 {
    grid-template-columns: repeat(4, 1fr);
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 20px;
  color: black;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 20px;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <p
      className="emotion-1"
    >
      Grade
    </p>
    <span
      className="emotion-3"
    >
      -
    </span>
  </div>
  <div
    className="emotion-1"
  >
    <p
      className="emotion-1"
    >
      Distance
    </p>
    <span
      className="emotion-6"
    >
      -
    </span>
  </div>
  <div
    className="emotion-1"
  >
    <p
      className="emotion-1"
    >
      Elevation
    </p>
    <span
      className="emotion-6"
    >
      -
    </span>
  </div>
  <div
    className="emotion-1"
  >
    <p
      className="emotion-1"
    >
      Time
    </p>
    <span
      className="emotion-6"
    >
      -
    </span>
  </div>
</div>
`;

exports[`ElevationSlice renders values 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: grid;
  grid-gap: 8px;
  grid-template-columns: repeat(2, 1fr);
  padding: 10px;
  border-left-color: mutedAccent;
  border-left-style: solid;
  border-left-width: 1px;
  border-right-color: mutedAccent;
  border-right-style: solid;
  border-right-width: 1px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    grid-template-columns: repeat(4, 1fr);
  }
}

@media screen and (min-width: 52em) {
  .emotion-0 {
    grid-template-columns: repeat(4, 1fr);
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 20px;
  color: orange;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 20px;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <p
      className="emotion-1"
    >
      Grade
    </p>
    <span
      className="emotion-3"
    >
      4.0%
    </span>
  </div>
  <div
    className="emotion-1"
  >
    <p
      className="emotion-1"
    >
      Distance
    </p>
    <span
      className="emotion-6"
    >
      100 miles
    </span>
  </div>
  <div
    className="emotion-1"
  >
    <p
      className="emotion-1"
    >
      Elevation
    </p>
    <span
      className="emotion-6"
    >
      50 feet
    </span>
  </div>
  <div
    className="emotion-1"
  >
    <p
      className="emotion-1"
    >
      Time
    </p>
    <span
      className="emotion-6"
    >
      20:00
    </span>
  </div>
</div>
`;
